tests/testthat/test-rule-fit-regression.R

```diff
@@ -157,162 +157,64 @@ test_that("tidy method - regression", {
-  })
+  expect_true( is.null(rf_fit_1$fit$xgb$best_iteration))
+  expect_true(!is.null(rf_fit_2$fit$xgb$best_iteration))
+  expect_true(!is.null(rf_fit_3$fit$xgb$best_iteration))
+})
 
-  expect_equal(
-    extract_fit_engine(pars_fit_4)$xgb$params$colsample_bytree,
-    1
-  )
+test_that("xrf_fit is sensitive to glm_control", {
+  rf_mod <-
+    rule_fit(trees = 3) %>%
+    set_engine("xrf", glm_control = list(type.measure = "deviance", nfolds = 8)) %>%
+    set_mode("regression")
 
-  # supply a proportion when count expected
-  expect_snapshot_error({
-    pars_fit_5 <-
-      rule_fit(mtry = .5, trees = 5) %>%
-      set_engine("xrf") %>%
-      set_mode("regression") %>%
-      fit(
-        Sale_Price ~ Neighborhood + Longitude + Latitude +
-          Gr_Liv_Area + Central_Air,
-        data = ames_data$ames
-      )
-  })
-
-  # supply a count when proportion expected
-  expect_snapshot_error({
-    pars_fit_6 <-
-      rule_fit(mtry = 3, trees = 5) %>%
-      set_engine("xrf", counts = FALSE) %>%
-      set_mode("regression") %>%
-      fit(
-        Sale_Price ~ Neighborhood + Longitude + Latitude +
-          Gr_Liv_Area + Central_Air,
-        data = ames_data$ames
-      )
-  })
-
-  expect_warning({
-    pars_fit_7 <-
-      rule_fit(trees = 5) %>%
-      set_engine("xrf", colsample_bytree = .5) %>%
-      set_mode("regression") %>%
-      fit(
-        Sale_Price ~ Neighborhood + Longitude + Latitude +
-          Gr_Liv_Area + Central_Air,
-        data = ames_data$ames
-      )},
-    "manually modified and were removed: colsample_bytree."
+  expect_error_free(
+    rf_fit_1 <- fit(rf_mod, mpg ~ ., data = mtcars)
   )
 
-  expect_equal(
-    extract_fit_engine(pars_fit_7)$xgb$params$colsample_bytree,
-    1
-  )
+  rf_fit_1_call_args <- rlang::call_args(rf_fit_1$fit$glm$model$call)
 
-  # supply both feature fraction and mtry
-  expect_snapshot({
-    pars_fit_8 <-
-      rule_fit(mtry = .5, trees = 5) %>%
-      set_engine("xrf", colsample_bytree = .5) %>%
-      set_mode("regression") %>%
-      fit(
-        Sale_Price ~ Neighborhood + Longitude + Latitude +
-          Gr_Liv_Area + Central_Air,
-        data = ames_data$ames
-      )},
-    error = TRUE
-  )
+  expect_equal(rf_fit_1_call_args$nfolds, 8)
+  expect_equal(rf_fit_1_call_args$type.measure, "deviance")
+})
 
-  expect_warning({
-    pars_fit_9 <-
-      rule_fit(mtry = 5, trees = 5) %>%
-      set_engine("xrf", colsample_bytree = .5) %>%
-      set_mode("regression") %>%
-      fit(
-        Sale_Price ~ Neighborhood + Longitude + Latitude +
-          Gr_Liv_Area + Central_Air,
-        data = ames_data$ames
-      )},
-    "manually modified and were removed: colsample_bytree."
-  )
+test_that("xrf_fit guards xgb_control", {
+  rf_mod <-
+    rule_fit(trees = 3) %>%
+    set_engine("xrf", xgb_control = list(nrounds = 3)) %>%
+    set_mode("regression")
 
-  expect_equal(
-    extract_fit_engine(pars_fit_9)$xgb$params$colsample_bytree,
-    1
+  expect_warning(
```

Comment:
This should use `expect_snapshot()`

Reply:
Agreed! Thanks for these two nudges.

I'm seeing:

```
── Error (test-rule-fit-regression.R:184:3): early stopping works in xrf_fit ───
Error in `!cran`: invalid argument type
Backtrace:
 1. testthat::expect_snapshot(...)
      at test-rule-fit-regression.R:184:2
 2. testthat:::expect_snapshot_helper(...)

── Error (test-rule-fit-regression.R:216:3): xrf_fit guards xgb_control ────────
Error in `!cran`: invalid argument type
Backtrace:
 1. testthat::expect_snapshot(...)
      at test-rule-fit-regression.R:216:2
 2. testthat:::expect_snapshot_helper(...)

[ FAIL 2 | WARN 0 | SKIP 0 | PASS 30 ]
```

when I try to generate these snapshots. Can you confirm that you also see this? I've generally ran into a few oddities with the testing environment with this package—I'm also unable to run the `make_*` data helpers interactively.

Reply:
I'm not able to reproduce this issue :( It works flawlessly for me 🤞